core/src/main/scala/pureconfig/BasicReaders.scala

```diff
@@ -165,30 +165,30 @@ trait NumericReaders {
 trait TypesafeConfigReaders {
 
   implicit val configConfigReader: ConfigReader[Config] =
-    ConfigReader.fromCursor(_.asObjectCursor.right.map(_.value.toConfig))
+    ConfigReader.fromCursor(_.asObjectCursor.right.map(_.objValue.toConfig))
 
   implicit val configObjectConfigReader: ConfigReader[ConfigObject] =
-    ConfigReader.fromCursor(_.asObjectCursor.right.map(_.value))
+    ConfigReader.fromCursor(_.asObjectCursor.right.map(_.objValue))
 
   implicit val configValueConfigReader: ConfigReader[ConfigValue] =
-    ConfigReader.fromCursor(c => Right(c.value))
+    ConfigReader.fromCursor(_.asConfigValue)
 
   implicit val configListConfigReader: ConfigReader[ConfigList] =
-    ConfigReader.fromCursor(_.asListCursor.right.map(_.value))
+    ConfigReader.fromCursor(_.asListCursor.right.map(_.listValue))
 
+  // TODO: improve error handling
```

Comment:
Could I trouble you to add to the comment a description of how you want to improve error handling?

Reply:
Sure, done 👍